File: clients/js/src/features/resolve_redirect.ts
import { ClientConfigBuilder, PolywrapClient } from "@polywrap/client-js";

export async function runTestCase(input: unknown): Promise<void> {
  if (!input || typeof input !== "object") {
    throw Error(
      "resolve_redirect test case input must be an object"
    );
  }

  const { from, to } = input as any;

  if (typeof from !== "string") {
    throw Error(
      "resolve_redirect input.from must be a string"
    );
  }

  if (typeof to !== "string") {
    throw Error(
      "resolve_redirect input.to must be a string"
    );
  }

  const config = new ClientConfigBuilder()
    .addRedirect(from, to)
    .build();

  const client = new PolywrapClient(config);

  console.log("Resolving Redirect");

  const result = await client.tryResolveUri({ uri: from });

  if (result.ok && result.value.type === "uri") {
    console.log(`Received URI '${result.value.uri}'`);
    console.log("Success!");
  }
}

File: clients/py/src/features/resolve_redirect.py
from typing import Any

from polywrap import PolywrapClient, PolywrapClientConfigBuilder, Uri
from pydantic import BaseModel, Field


class TestCaseInput(BaseModel):
    from_: str = Field(..., alias="from")
    to: str


def run_test_case(input: Any) -> None:
    input_obj = TestCaseInput.parse_obj(input)

    from_uri = Uri.from_str(input_obj.from_)
    to_uri = Uri.from_str(input_obj.to)

    config = PolywrapClientConfigBuilder().set_redirect(from_uri, to_uri).build()

    client = PolywrapClient(config)

    print("Resolving Redirect")

    result = client.try_resolve_uri(from_uri)

    match result:
        case Uri():
            print(f"Received URI '{result.uri}'")
            print("Success!")
        case _:
            print("Failed!")
